src/Rasa.Game/Structures/Entity.cs

```diff
@@ -0,0 +1,16 @@
+﻿namespace Rasa.Structures
+{
+    using Data;
+
+    public abstract class Entity
+    {
+        public EntityGUID EntityGUID { get; set; }
+        public ulong EntityId => EntityGUID.Id;
+        public EntityType EntityType { get; set; }
```

Comment:
Why is this here? Should this value come from the EntityGUID instance too like EntityId?

Reply:
I changed this, but I'm not seeing the point of encoding the type if I'm just keeping track in the property too. It looks weird. Thoughts?